giturlparse/platforms/bitbucket.py

```diff
@@ -4,7 +4,7 @@
 class BitbucketPlatform(BasePlatform):
     PATTERNS = {
         "https": (
-            r"(?P<protocols>(git\+)?(?P<protocol>https))://(?P<_user>.+)@(?P<domain>.+?)"
+            r"(?P<protocols>(git\+)?(?P<protocol>https))://(?P<_user>(.+)?)@(?P<domain>.+?)"
```

Comment:
this does not make it optional... I think you meant

```suggestion
            r"(?P<protocols>(git\+)?(?P<protocol>https))://((?P<_user>.+)@)?(?P<domain>.+?)"
```

but with that -- there better be a test added to ensure correct/expected operation

PS I have no relation to development or use of this package

Reply:
Oh, you're right, and I just noticed the `@` would need to be made optional as well. Good catch, thanks for reading the patch critically :)

Reply:
Nevermind, I misread the parenthesis in your change. You did make it optional, sorry!